include/pagmo/utils/discrepancy.hpp

```diff
@@ -81,7 +82,8 @@ inline std::vector<double> sample_from_simplex(std::vector<double> in)
         pagmo_throw(std::invalid_argument, "Input vector must have all elements in [0,1]");
     }
     if (in.size() > 0u) {
-        std::sort(in.begin(), in.end());
+        std::sort(in.begin(), in.end(),
+                  [](const double &el1, const double &el2) { return detail::less_than_f(el1, el2); });
```

Comment:
Doesn't it work here if, instead of a lambda, you use ``detail::less_than_f<double>`` directly?